obelisk/cpp/obelisk_cpp/include/obelisk_controller.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include "obelisk_node.h"
+
+namespace obelisk {
+
+    template <typename ControlMessageT, typename EstimatorMessageT> class ObeliskController : public ObeliskNode {
+      public:
+        explicit ObeliskController(const std::string& name, const std::string& ctrl_key = "pub_ctrl",
+                                   const std::string& est_key = "sub_est", const std::string& timer_key = "timer_ctrl")
+            : ObeliskNode(name), ctrl_key_(ctrl_key), est_key_(est_key), timer_key_(timer_key) {
+
+            // Register all components
+            this->RegisterTimer("timer_ctrl_setting", timer_key_, std::bind(&ObeliskController::ComputeControl, this));
+            this->RegisterPublisher<ControlMessageT>("pub_ctrl_setting", ctrl_key_);
+            this->RegisterSubscription<EstimatorMessageT>(
+                "sub_est_setting", est_key_, std::bind(&ObeliskController::UpdateXHat, this, std::placeholders::_1));
+        }
+
+        /**
+         * @brief Configures all the required ROS components. Specifcially this
+         * registers the control_publisher_, state_estimator_subscriber_, and the
+         * control_timer_. Also makes a call to ObeliskNode on configure to parse
+         * and create the callback group map.
+         *
+         * @param prev_state the state of the ros node.
+         * @return success if everything completes.
+         */
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual on_configure(
+            const rclcpp_lifecycle::State& prev_state) {
+            ObeliskNode::on_configure(prev_state);
+
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
```

Comment:
Do we need to implement this? It's just one super call. On the python side, we don't even have this and calling super just straight up calls the base class' `on_configure`. We can probably resolve this at the same time as #35.

Reply:
Yea I left this in there until we resolve #35 since I don't want to delete it just to re-write it. I think I'll go ahead and implement #35 soon and that will just clear a lot of this up.